### ISSUE TITLE
[FEATURE] Add beginner category

### ISSUE THREAD

**Issue by fejes713** [Chalarangelo/30-seconds-of-code#640]
### Description 
We have over 200 snippets on our website. Yes, they're very useful in pretty much every situation! However, when a user visits our site he is welcomed with `ary `, `call`, `collectInto` snippets. They're very informative and useful but beginners won't find them `friendly?` - they might just debounce of the website and won't give __30secondsofcode__ another chance. Why don't we show some `interesting and easy to understand` snippets first or simply mark few snippets as `beginner friendly`?

### Possible Solution
I suggest that we should implement beginner tags in our `tag` database and show those snippets first when someone visits the website. That way we are definitely going to help new developers learn faster and stick to using __30secondsofcode__.

**Comment by Chalarangelo** [comment]
I agree that we should approach this a bit differently than we have in the past. However, recategorizing the snippets in `beginner` and `advanced` etc is only half a solution. I was actually thinking of a deeper overhaul of the website:

### My suggestion for the website

- Create a clean and lean landing page, so that people can read a short description of the project and what it has to offer. On the same page, we can add a *snippet of the day* section, built nightly from the cron job, that will pick a random snippet to feature daily. That way we can keep visitiors interested for longer and help people discover interesting snippets.
- Add some proper navigation and separate the content into pages. Each major tag that we currently have will be its own page, so that we can keep load times shorter. However, the main navigation sidebar should remain the same as it is, allowing for the website to feel as a single entity.
- Add a page for beginners that is featured promimently in the landing page links. Curate the beginner-friendly snippets a bit more carefully and add some more content to that page, such as extra links to documentation and learning resources to encourage people to start learning from our snippet database.
- Add a page for the archived snippets, just so that people can more easily find them outside of the repo.
- Add pages similar to the beginner page for certain categories of tasks (not sure if viable/applicable). For example, we could add a page with snippets for working with a specific kind of project. My reasoning for this is that we can curate single pages with a main goal more carefully and add more links and resources to them.

In short, the current website is something I put together very quickly and then we just incrementally improved it until it sort of worked and left it there. We really should work on improving it and making it more usable, while improving our branding and overall visibility. After all, we stil have the README if we want everything displayed as-is without any changes or extra features (and it serves a great purpose as-is). 

Opinions are, as always, appreciated. Let's figure out the best approach to this and start building it.

**Comment by kriadmin** [comment]
>On the same page, we can add a snippet of the day section, built nightly from the cron job, that will pick a random snippet to feature daily

Can we have a voting system which decides snippet of the day? Users can vote for the snippets that helped them. We can even have a login feature for a personalized experience.

**Comment by Chalarangelo** [comment]
Without any backend, neither of these things is possible, @kriadmin. For the highest exposure of snippets we should use a random rotation and have 4-5 snippets each day, so that users who visit the page multiple times a day or refresh can see a few different snippets. That way it should take about 2 months to run through all of them and they will all be featured at one time or another.

**Comment by Rob-Rychs** [comment]
maybe a multi-tag system? 
In the left hand-menu underneath the search bar it would be nice if there was a grouping of three buttons that switched the order/categorization of the snippets
1. difficulty (default) -> rearranges snippets by alphabetic categorized under Beginner / intermediate / advanced
2. category -> rearranges snippets into current categories
3. a-z -> rearrange the snippets alphabetically like it's sesame street
Both the order of snippets in main viewport and menu should toggle/rearrange by selection

**Comment by fejes713** [comment]
@Chalarangelo @kriadmin That would definitely require a backend which is not urgent right now. Something similar was suggested in __30 seconds of CSS__ and we delayed the idea a bit.

Right now I think the best solution is what @Chalarangelo suggested. We should discuss the details on `Gitter` these days. I am free for few days; @Chalarangelo when do you want us to start this? Tomorrow?

**Comment by flxwu** [comment]
I also agree with @Chalarangelo 's ideas.

**Comment by Chalarangelo** [comment]
Hey, everyone. I am going to be available starting Friday up to Monday, so we can start implementing these changes and give the website a facelift. I am going to play around with the design of the landing page until Friday, so we have a basis to start from, then we can move on to the rest of the scripting and tools. I believe this is a procedure that will take just a couple of days, so we can all pitch in during the weekend and get it done.

I'll soon start a branch with the new landing page, so we can get started.

**Comment by fejes713** [comment]
@Chalarangelo be sure to reach out to me on Friday. I would like to help.

**Comment by skatcat31** [comment]
I think we should probably update archived snippets then as Academic. At that point we can go into heavier detail about EXISTING concepts and unroll some snippets for longer discussion and have it be more of a "learn for the sake of learning" section than the snippets, which are "cool tricks and fast tips".

After all that'd be a great place to put in something like a server side event channel system with subscribers that can be used to create distributed messaging systems with cascading messages. 100 lines of code, and it's a feature a LOT of people would love

**Comment by fejes713** [comment]
@skatcat31 I would love to see that. Few principles that take more than 30 seconds to understand but are used in modern workflows.

**Comment by Chalarangelo** [comment]
For clarity and transparency, here's the list of beginner snippets that we will be featuring in the beginner page:

- everyNth
- filterNonUnique
- last
- maxN
- minN
- nthElement
- sample
- similarity
- tail
- currentURL
- hasClass
- getMeridiemSuffixOfInteger
- factorial
- fibonacci
- gcd
- isDivisible
- isEven
- isPrime
- lcm
- randomIntegerInRange
- sum
- reverseString
- truncateString

Special thanks to @fejes713 for hand-picking these.

**Comment by fejes713** [comment]
@Chalarangelo List could be improved, it requires nearly no effort to update it with the newest `web.js` script.

**Comment by Chalarangelo** [comment]
@fejes713 Yep, we could alter the list of beginner snippets anytime. The fact that it can be customized with nearly no effort is a big advantage, as beginners looking to make their first PR could actually tell us which snippets they find most useful. 😉

**Comment by Chalarangelo** [comment]
Resolved in #642
